### PR TITLE
Convert script version ID capnp representation to UUID.

### PR DESCRIPTION
**Pull request by jp4a50**
We want to constrain the ID to be a UUID so let's enforce that constraint in the capnp interface.

### CODE REVIEW

File: src/workerd/io/script-version.capnp [hunk 1]
```diff
@@ -8,10 +8,15 @@ using Cxx = import "/capnp/c++.capnp";
...
 struct ScriptVersion {
-  id @0 :Text;
-  # An ID that should be used to uniquely identify this version.
-  tag @1 :Text;
+  id :group {
+    upper @0 :UInt64;
+    # Most significant bits of the UUID.
+    lower @1 :UInt64;
+    # Least significant bits of the UUID.
+  }
```
Comment (dom96) [inline src/workerd/io/script-version.capnp:16]: Is this struct unused in production? if it is used, couldn't we end up with errors when the old struct layout is parsed while this new one is expected?

**Review by dom96 — APPROVED** [review]
One comment, otherwise LGTM.

File: src/workerd/io/script-version.capnp (hunk 1 above)
Comment (taylorlee) [reply src/workerd/io/script-version.capnp:16]: This has never been used in production, it's safe to change.

**Review by taylorlee — APPROVED** [review]

**Review by jasnell — APPROVED** [review]
Not sure `UUIDToString` is a great name for this but don't have a good alternative. Overall LGTM

**Pull request merged**